### PR TITLE
First tests

### PR DESCRIPTION
**Pull request by Liljenmaa**

### CODE REVIEW

File: dummy.test.js [hunk 1]
```diff
@@ -1,3 +1,11 @@
...
+
+test("countBy works", () => {
+  expect(countBy([
+     { 'user': 'barney', 'active': true },
+     { 'user': 'betty', 'active': true },
+     { 'user': 'fred', 'active': false }
+   ], value => value.active)).toBe(
+    { 'true': 2, 'false': 1 }
+  );
 });
```
Comment (II-KA) [inline dummy.test.js]: These tests are duplicated in countBy.test.js. This dummy file can probably be deleted since the test setup is now working.

File: test/countBy.test.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+test("countBy works", () => {
+    expect(countBy([
+        { 'user': 'barney', 'active': true },
+        { 'user': 'betty', 'active': true },
+        { 'user': 'fred', 'active': false }
+    ], value => value.active)).toBe(
+        { 'true': 2, 'false': 1 }
+    );
+});
```
Comment (II-KA) [inline test/countBy.test.js]: I'd recommend using describe-it "syntax" to group the tests (in all test files), since we are going to have multiple tests for each utility function. `It()` can be used to create new tests and `describe()` can be used to further group the tests if needed. The top level describe should contain the function name 👍 

Additionally, separating the users array to a variable makes the test easier to read in my opinion 

```suggestion
describe("countBy", () => {
  it("correctly counts active users", () => {
    const users = [
      { user: "barney", active: true },
      { user: "betty", active: true },
      { user: "fred", active: false },
    ];
    expect(countBy(users, (value) => value.active)).toBe({ true: 2, false: 1 });
  });
});
```

**Review by II-KA — CHANGES_REQUESTED** [review]

File: test/map.test.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import map from "../src/map.js";
+
+function square(n) {
+  return n * n;
+}
+
+describe("map", () => {
+  it("maps numbers to squared", () => {
+    expect(map([4, 8], square)).toIncludeAllMembers([16, 64]);
+  });
```
Comment (II-KA) [inline test/map.test.js]: Looks like `toBe` does not compare arrays correctly. Changed to https://jest-extended.jestcommunity.dev/docs/matchers/Array#toincludeallmembersmembers and the test passes

**Comment by II-KA** [comment]
Btw the command `npx jest <filename>` can be used to run individual test files. Running all of them in development might get annoying since multiple fail

File: test/words.test.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    expect(words('fred, barney, & pebbles')).toBe(
+        ['fred', 'barney', 'pebbles']
+    );
+});
+
+test("words with regex", () => {
+    expect(words('fred, barney, & pebbles', /[^, ]+/g)).toBe(
+        ['fred', 'barney', '&', 'pebbles']
+    );
+});
```
Comment (II-KA) [inline test/words.test.js]: Using toIncludeAllMembers will make the tests pass

```suggestion
test("words without regex", () => {
    expect(words('fred, barney, & pebbles')).toIncludeAllMembers(
        ['fred', 'barney', 'pebbles']
    );
});

test("words with regex", () => {
    expect(words('fred, barney, & pebbles', /[^, ]+/g)).toIncludeAllMembers(
        ['fred', 'barney', '&', 'pebbles']
    );
});
```

File: test/map.test.js (hunk 3 above)
Comment (Liljenmaa) [reply test/map.test.js]: Or should it be "toStrictEqual"?

File: test/map.test.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import map from "../src/map.js";
+
+describe("map", () => {
+    it("regards null as empty array", () => {
+        const square = (n) => n * n;
+
+        expect(map(null, square)).toIncludeAllMembers(
+            []
+        );
```
Comment (II-KA) [inline test/map.test.js]:
```suggestion
        expect(map(null, square)).toBeArrayOfSize(0)
```

File: test/map.test.js (hunk 3 above)
Comment (II-KA) [reply test/map.test.js]: Could also be ok, but since we are using jest-extended lets prefer that. It also has a lot of matchers for object types

File: test/countBy.test.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+describe("countBy", () => {
+    it("correctly counts active users", () => {
+        const users = [
+            { 'user': 'barney', 'active': true },
+            { 'user': 'betty', 'active': true },
+            { 'user': 'fred', 'active': false },
+        ];
+        expect(countBy(users, (value) => value.active)).toBe({ true: 2, false: 1 });
+    });
+});
```
Comment (II-KA) [inline test/countBy.test.js]: I'd recommend adding tests for some corner cases e.g. no user is active

File: test/isObjectLike.test.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import isObjectLike from "../src/isObjectLike.js";
+
+describe("isObjectLike", () => {
```
Comment (II-KA) [inline test/isObjectLike.test.js]: How about testing a date?

File: test/toNumber.test.js [hunk 8]
```diff
@@ -0,0 +1,146 @@
+import toNumber from "../src/toNumber.js";
+
+describe("toNumber", () => {
```
Comment (II-KA) [inline test/toNumber.test.js]: I'd add more describes inside this to group the `it()` tests

```
describe('transforms to itself', () => {});
describe('transforms to number', () => {});
describe('transforms to not a number', () => {});
...
```

File: test/countBy.test.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
+import countBy from "../src/countBy.js";
+
+describe("countBy", () => {
+    it("can correctly count attributes from an array of objects", () => {
+        const users = [
+            { user: 'barney', active: true },
+            { user: 'betty', active: true },
+            { user: 'fred', active: false },
+        ];
```
Comment (II-KA) [inline test/countBy.test.js]: Users array is used in multiple places, so it could be defined once in the beginning
```suggestion
const users = [
  { user: 'barney', active: true },
  { user: 'betty', active: true },
  { user: 'fred', active: false },
];

describe("countBy", () => {
    it("can correctly count attributes from an array of objects", () => {
```

**Comment by II-KA** [comment]
Rebased on main
Refactored tests:
- Group it()-statements with describe() and rename test descriptions
- Define commonly used variables in the beginning of the file
